src/puptoo/app.py

```diff
@@ -71,6 +72,19 @@ def handle_signal(signal, frame):
     global running
     running = False
 
+def redis_client() :
+    return Redis(host=config.REDIS_HOST, port=config.REDIS_PORT) 
+
+def handle_retries(redis, request_id):
+    count = redis.get(request_id)
+    if not count:
+        count = 0
+    if int(count) == 3:
+        raise Exception("Message process attempts exceeded")
```

Comment:
It would be cool to log the request ID here as well. "Message process attempts exceeded for {request_id}"

Reply:
Done!